ts/screens/profile/TosScreen.tsx

```diff
@@ -38,23 +46,36 @@ const TosScreen = () => {
     setIsLoading(true);
   }, [setIsLoading]);
 
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      header: () => (
+        <HeaderSecondLevel
+          type="singleAction"
+          title={I18n.t("profile.main.privacy.title")}
+          backAccessibilityLabel={I18n.t("global.buttons.back")}
+          goBack={navigation.goBack}
+          firstAction={{
+            icon: "help",
+            onPress: startSupportRequest,
+            accessibilityLabel: I18n.t(
+              "global.accessibility.contextualHelp.open.label"
+            )
+          }}
+        />
+      )
+    });
+  }, [navigation, startSupportRequest]);
+
```

Comment:
Since there is a lot of boilerplate and for each screen in my opinion it can be really stressful and it can bring a regression to the developer experience, what do you think to create a custom hook such as `useHeaderSecondLevel` which basically accept all the properties of the `HeaderSecondLevel` component but some attributes are optional? In this way, integrating an header can be more simple and the code is more reusable without a big boilerplate.

I would imagine an hook defined something like this:
```typescript

type HeaderSecondLevelHookCommonProps = {
  // ...(same HeaderSecondLevelProps with only title and type as required ...
  // ... since the most of the time backAccessibilityLabel is always ...
  // ... "back", and goBack is navigation.goBack at least the developer doesn't want to customize it), 
};

interface HeaderWithSupport extends HeaderSecondLevelHookCommonProps {
  supportRequest: true;
  firstAction?: never;
  faqCategories: Array<string>;
}

interface HeaderWithoutSupport extends HeaderSecondLevelHookCommonProps {
  supportRequest?: false;
}

export type HeaderSecondLevelHookProps = HeaderWithSupport | HeaderWithoutSupport;

const useHeaderSecondLevel = ({
  type,
  title,
  backAccessibilityLabel,
  goBack,
  firstAction,
  supportRequest,
  faqCategories,
  // ...allTheOtherHeaderProps,
}: HeaderSecondLevelHookProps) => {

  const startSupportRequest = useStartSupportRequest({
    faqCategories: supportRequest ? faqCategories : [],
    contextualHelpMarkdown
  });

  const navigation = useNavigation();

  useLayoutEffect(() => {
    navigation.setOptions({
      header: () => (
        <HeaderSecondLevel
          type={type}
          title={title}
          backAccessibilityLabel={backAccessibilityLabel || I18n.t("global.buttons.back")}
          goBack={goBack ? goBack : navigation.goBack}
          firstAction={supportRequest ? {
            icon: "help",
            onPress: startSupportRequest,
            accessibilityLabel: I18n.t(
              "global.accessibility.contextualHelp.open.label"
            )
          } : firstAction}
          //...allTheHeaderProps
        />
      )
    });
  }, [navigation, startSupportRequest]);
}
```


And then into `PrivacyMainScreen` could become:
```typescript
useHeaderSecondLevel({
  type: "singleAction"
  title: I18n.t("profile.main.privacy.title")
  supportRequest: true,
  faqCategories: ["privacy"]
})
```

Instead, for simple screens where you don't even need to handle contextual help and support, it could be simply 4 lines of code:
```typescript
useHeaderSecondLevel({
  type: "singleAction"
  title: I18n.t("profile.main.privacy.title")
})
```
Since in the most of our screens there is the contextual help button, i think that using a `supportRequest` boolean attribute as an optional attribute and if it is true, automatically set the `firstAction` with the icon help button, the same also for the `goBack`, since the most of the time is a `navigation.goBack()`, setting it as default can be less verbose, the same for `backAccessibilityLabel` that the most of the time is always `I18n.t("global.buttons.back")`. Of course we are just setting default values but nothing prevents the developer from setting different values

Reply:
Great Idea @Hantex9 going to implement it

Reply:
Even if your suggestion was interesting it would bring some difficult cases to handle:
- the position of the help action could change based on the number of the action supported by the header
- the typedef of the hook props would be even more difficult to specify/comprehend than the header ones

for these reason I think to have a common hook would be a great idea, but instead of handling all the scenarios above described, this hook should solve only the base scenario (the most used inside app code) header with or without help desk button.

[42cc79b](https://github.com/pagopa/io-app/pull/5028/commits/42cc79bfc93b222236ce5cded9679e727c86927e) has an implementation of it, what do you think?

Reply:
That's really useful! I agree with you and to avoid this scenario the implemented custom hook that handles the two more used case is a good solution to improve the DX! 🙌

Reply:
@CrisTofani @Hantex9 Just a design note. Regardless of the other actions, the help one should be always the last starting from the left.